crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs

```diff
@@ -844,4 +879,24 @@ impl<'ast> ScopeHoistingFinalizer<'_, 'ast> {
       },
     );
   }
+
+  fn process_fn_decl(&mut self, decl: &mut allocator::Box<'_, ast::Function<'_>>) -> Option<()> {
+    if !self.ctx.options.keep_names {
+      return None;
+    }
+    let id = decl.id.as_ref()?;
+    let symbol_id = id.symbol_id.get()?;
+    let symbol_ref = (self.ctx.id, symbol_id).into();
+    dbg!(&self.ctx.renamed_symbol_map);
+    if let Some(original_name) = self.ctx.renamed_symbol_map.get(&symbol_ref) {
```

Comment:
Does this do the same work as `renamed_symbol_map`? 
```rust
let is_changed = symbol_ref.name() == self.ctx.symbol_db.canonical_name_for(symbol_ref);
```

Reply:
the same effect but should be faster, `canonical_name_for` needs two steps, first get the finalized name, then get the deconflicted name, get symbol also need extra lookup 1 step vs 3 step.